server/src/main/java/org/elasticsearch/action/datastreams/options/PutDataStreamOptionsAction.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package org.elasticsearch.action.datastreams.options;
```

Comment:
Is there a reason that these actions are in the server project but the rest of the code for them is in the data stream module?

Reply:
Hm good point, this was a bit premature. We have security specific lifecycle tests that need the actions to be in the server to access them. As I was using the lifecycle as a model, I placed them here too but since we do not need this yet I will move it to the data streams modules. Good catch